deploy/basic/terraform/variables.tf

```diff
@@ -192,3 +192,80 @@ locals {
   compute_shape_description              = lookup(local.compute_shape_flexible_vs_descriptions, local.instance_shape, local.instance_shape)
   lb_shape_flexible                      = "flexible"
 }
+
+# Additional stuff for extra demo (DNS, Cert, WAF, Traffic)
+# DNS Entries
+variable "enable_dns_zone" {
+  default = false
+}
+
+variable "dns_lb_entry" {
+  default = "internal-lb"
+}
+
+variable "dns_waf_entry" {
+  default = "store"
+}
+
+locals {
+  dns_lb_domain = "${var.dns_lb_entry}.${var.dns_zone_name}"
+  dns_waf_domain = "${var.dns_waf_entry}.${var.dns_zone_name}"
+}
+
+# Certificate
+variable "certificate_certificate_name" {
+  default = "mushop-lets-encrypt-certificate"
+}
+
+variable "enable_acme_certificate" {
+  default = false
+}
+
+variable "acme_email" {
+  default = ""
+}
+
+variable "acme_server_url" {
+  default = "https://acme-staging-v02.api.letsencrypt.org/directory"
+}
+
+variable "dns_zone_name" {}
+
+variable "enable_waf" {
+  default = "false"
+}
+
+# CAPTCHA variables
+
+variable "enable_captcha_challenge" {
+  default = "false"
+}
+
+variable "waas_policy_waf_config_captchas_failure_message" {
+  default = "Incorrect CAPTCHA"
+}
+
+variable "waas_policy_waf_config_captchas_session_expiration_in_seconds" {
+  default = "3600"
+}
+
+variable "waas_policy_waf_config_captchas_submit_label" {
+  default = "Show me the Products!"
+}
+
+variable "waas_policy_waf_config_captchas_title" {
+  default = "Quickly before you browse products..."
+}
+
+variable "waas_policy_waf_config_captchas_url" {
+  default = "/product.html"
+}
+
+variable "waas_policy_waf_config_captchas_footer_text" {
+  default = "Enter the letters and numbers as they are shown in the image above."
+}
+
+variable "waas_policy_waf_config_captchas_header_text" { 
+  default = "Help us keep pet product shopping safe please."
```

Comment:
Have you tested the MuShop SPA with CAPTCHA enabled? The SPA makes frequent `XHR` requests to the server for `*.html` files (views) during navigation. If one were to return the redirect for CAPTCHA, it might not be handled properly

Reply:
I did some tests with CAPTCHA and it appeared to only ask me once per browser session.  In reality I could have chosen any advanced feature to try.